server/rest/src/main/java/org/infinispan/rest/resources/MetricsResource.java

```diff
@@ -31,11 +35,20 @@
  */
 public final class MetricsResource implements ResourceHandler {
 
+   private final static Log log = LogFactory.getLog(MetricsResource.class, Log.class);
+
    private static final String METRICS_PATH = "/metrics";
 
    private final MetricsRequestHandler requestHandler = new MetricsRequestHandler();
 
    public MetricsResource() {
+      try {
+         new JmxRegistrar().init();
+      } catch (IOException | IllegalArgumentException e) {
+         // Smallrye uses a global singleton registry which is a nightmare for tests where more than one
+         // server has to exist in a single JVM. It fails and we can't do anything about it.
```

Comment:
Isn't the UUID in `CacheManagerMetricsRegistration` supposed to fix this?

If a test sets a node name and sets the same name for 2 nodes (or 2 different tests set the same name) then the test should be fixed.

Reply:
Metrics produced by JmxRegistrar do not have that tag because ... we're using an internal smallrye utility class which was not created for public consumption so it's not that flexible. And quite frankly such a unique tag would just allow us to (wrongly) register duplicates of metrics that are really just JVM monitoring data. That's what JmxRegistrar produces. Once is enough. All further attempts will fail. Benign error -> ignore. It's even possible that another app from the jvm/jee container already executed JmxRegistrar. I thought of maybe having a global flag protected by synchronization and try to do this exactly once. Not sure it's worth the hassle because would only provide mutual exclusion between our cache managers but not with other apps. Explicitly testing for presence of a metric and registering it only if not found is a source of race conditions. The MetricRegistry also does not provide any putIfAbsent semantics. Life sucks.

Reply:
IMO if it's ok for the registration to fail, then we shouldn't log it as an error, and the comment should mention why it's ok for it to fail.